_core/smart/functions.md

```diff
@@ -0,0 +1,151 @@
+    <td>Fetches the value associated with a given key in the map, or returns <code>'null</code> if there is none.</td>
+  </tr>
+  <tr>
+    <td><code>(set-entry! map-name key-tuple value-tuple)</code></td>
+    <td>Sets the value of key-tuple in the data map.</td>
+  </tr>
+  <tr>
+    <td><code>(insert-entry! map-name key-tuple value-tuple)</code></td>
+    <td>Sets the value of <code>key-tuple</code> in the data map if and only if an entry does not already exist.</td>
+  </tr>
+  <tr>
+    <td><code>(delete-entry! map-name key-tuple)</code></td>
+    <td>Removes the value associated with the input key for the given map.</td>
+  </tr>
+</table>
+
+Data maps make reasoning about functions easier. By inspecting a given function definition, it is clear which maps will be modified and, even within those maps, which keys are affected by a given invocation. Also, the interface of data maps ensures that the return types of map operations are fixed length; Fixed length returns is a requirement for static analysis of a contract's runtime, costs, and other properties.
+
+## List operations and functions
+
+Lists may be multi-dimensional. However, note that runtime admission checks on typed function-parameters and data-map functions like `set-entry!` are charged based on the _maximal_ size of the multi-dimensional list.
+
+You can call `filter` `map` and `fold` functions with user-defined functions (that is, functions defined with `(define ...)`, `(define-read-only ...)`, or `(define-public ...)`) or simple, native functions (for example, `+`, `-`, `not`).
+
+## Intra-contract calls
+
+A smart contract may call functions from other smart contracts using a `(contract-call!)` function:
+
+```cl
+(contract-call! contract-name function-name arg0 arg1 ...)
+```
+
+This function accepts a function name and the smart contract's name as input. For example, to call the function `token-transfer` in the smart contract, you would use:
+
+`(contract-call! tokens token-transfer burn-address name-price))`
+
+For intra-contract calls dynamic dispatch is not supported. When a contract is launched, any contracts it depends on (calls) must exist. Additionally, no cycles may exist in the call graph of a smart contract. This prevents recursion (and re-entrancy bugs. A static analysis of the call graph detects such structures and they are rejected by the network.
+
+A smart contract may not modify other smart contracts' data directly; it can read data stored in those smart contracts' maps. This read ability does not alter any confidentiality guarantees of Clarity. All data in a smart contract is inherently public, andis readable through querying the underlying database in any case.
+
+To read another contract's data, use `(fetch-contract-entry)` function. This behaves identically to `(fetch-entry)`, though it accepts a contract principal as an argument in addition to the map name:
+
+```cl
+(fetch-contract-entry
+  'contract-principal
```

Comment:
For now, it makes more sense for this to be `contract-name` rather than `contract-principle`, because at the moment, names are used rather than the contract principal.

Reply:
✔️ @kantai  this doesn't require a source change in the repo does it?

Reply:
I don't think so